autogen/agentchat/contrib/graph_rag/graph_rag_agent.py

```diff
@@ -0,0 +1,76 @@
+from typing import List, Protocol
+
+from autogen.agentchat import ConversableAgent
+
+from .document import Document
+from .graph_store import GraphStore
+
+
+class GraphRagAgent(ConversableAgent, Protocol):
+    """
+    A graph rag agent is a conversable agent which could query graph database for answers.
+
+    An implementing agent class would
+    1. create a graph in the underlying database with input documents
+    2. use the retrieve() method to retrieve information.
+    3. use the retrieved information to generate and send back messages.
+
+    For example,
+    graph_rag_agent = GraphRagAgent(
+        name="movie knowledge graph agent",
+        human_input_mode="NEVER",
+        max_consecutive_auto_reply=3,
+        retrieve_config={
+            "docs_path": [
+                "./data/movies.txt",
+            ],
+            "llm_config" = autogen.config_list_from_json("OAI_CONFIG_LIST")
+            "database_config" = {
+                "host": "127.0.0.1",
+                "port": 6379,
+                "table_name": "movies"
+            }
+        },
+    )
+
+    # initialize database (internally)
+    # self._init_db(input_doc=[Document(doc) for doc in retrieve_config["docs_path"]])
+
+    question = "Name a few actors who've played in 'The Matrix'"
+
+    answer = graph_rag_agent.retrieve(question)
```

Comment:
In general, we don't want to add new specific APIs for each agent, if existing APIs can be reused. Can we reuse `generate_reply` here?

Reply:
removed and updated the example.